### ISSUE TITLE
migrate build system to maven

### ISSUE THREAD

**Issue by carrot-garden** [restlet/restlet-framework-java#5]
Jerome, hi;

I know you answered this before ("too many editions, maven did not work on the first try in the past"), but this is is a recurring issue for me, so here you have it again :-)

My user story is like this:

I build for osgi embedded environments, like felix (not equinox/eclipse, like Brian does);

I use strictly maven driven builds on jenkins with eclipse 
interactive development support via m2e connectors;

I use "major" osgi maven plugins, such as
http://felix.apache.org/site/apache-felix-maven-bundle-plugin-bnd.html
http://karaf.apache.org/manual/2.2.5/developers-guide/features-maven-plugin.html
http://eclipse.org/m2e/
http://eclipse.org/tycho/

yes, there is a world of difference between manifest-first and pom.xml-first approaches;
but it does not meant they are not compatible; there are ways to make it work together (tycho);

the kind of hackery you do with pom.xml generation for your restlet builds makes me think you never use restlet in osgi/maven (non eclipse) builds so you do not even see the weird things I have to deal with when including restlet artifacts as dependencies in my projects :-)

bottom line is: both your manifest.mf and pom.xml bundle import/dependency definitions are useless for me; the best work around I found so far for inclusion of restlet and its extensions in my builds is to create custom osgi bundle projects which embed restlet, extensions, dependencies - with heavy duty import/export exclusion, version override, etc; pretty much the way spring http://ebr.springsource.com/repository/app/ does only worse :-)

I wish you would have time to re-try maven-centric build again, and add your much needed multi-edition builds as customization / maven plugins for restlet on top of that;

thank you;

Andrei.

**Comment by jlouvel** [comment]
Finally, Restlet Framework is now fully Maven native!
https://github.com/restlet/restlet-framework-java/discussions/1434
